From e571a5e92e7229d11ef9c75dd6b7437375e8e671 Mon Sep 17 00:00:00 2001
From: Vishnu Vardhan Bantanahal <quic_vishvard@quicinc.com>
Date: Thu, 1 Jun 2023 17:55:00 +0530
Subject: [PATCH] net: added a new netdevice priv flag for soft tunnel

This flag can be used by flow acceleration managers
to identify whether a flow is originated from or
destined to a tun/tap device

Change-Id: I9631a13f083f5096be2c0046eedd277b9971c72d
Signed-off-by: Amit Gupta <amitgupt@codeaurora.org>
Signed-off-by: Pavithra R <pavir@codeaurora.org>
Signed-off-by: Vishnu Vardhan Bantanahal <quic_vishvard@quicinc.com>
---
 drivers/net/tun.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 7c8db8f6f661..c766087be7fd 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -998,6 +998,7 @@ static int tun_net_init(struct net_device *dev)
 	dev->vlan_features = dev->features &
 			     ~(NETIF_F_HW_VLAN_CTAG_TX |
 			       NETIF_F_HW_VLAN_STAG_TX);
+	dev->priv_flags_ext |= IFF_EXT_TUN_TAP;
 
 	tun->flags = (tun->flags & ~TUN_FEATURES) |
 		      (ifr->ifr_flags & TUN_FEATURES);
-- 
2.34.1

